src/codec/frame.rs

```diff
@@ -12,6 +12,28 @@ use super::AsyncSocket;
 use crate::packet::connected::{FramesMut, FramesRef};
 use crate::packet::{unconnected, Packet};
 
+/// `Framed` is a base structure for socket communication.
+/// In this project, it wraps an asynchronous UDP socket and implements the
+/// [`Stream`](futures::stream::Stream) and [`Sink`](futures::sink::Sink) traits.
+/// This allows for reading and writing RakNet frames over the socket.
+/// It supports receiving both unconnected and connected packets, but can only send unconnected
+/// packets. This design separates the offline and online packet handlers.
+///
+/// Properties:
```

Comment:
The fields of non-public APIs may change, so there is less need to write documentation for them.

Reply:
You can use `//` directly above the field for simple comments.

Reply:
These documents write while reading source for `raknet-rs`, so they are not specifically crafted technical documents aimed at a public API, which may make them somewhat verbose. However, they might help others understand.